### ISSUE TITLE
Bug: Error when returning to previously completed file upload lesson 

### ISSUE THREAD

**Issue by terichadbourne** [ProtoSchool/protoschool.github.io#217]
This issue is a result of the way caching is implement, but it causes a particularly noticeable problem in file upload lessons. 

The way the system currently works is that when you load a lesson, we check whether there's cached code from a previous visit and the lesson is marked as having been passed previously, and if both are true, we load the cached code into the editor and run `run` to try to reproduce the appropriate success message and make the "Submit" button become a "Next" or "More Tutorials" button: 

```
    onMounted: function (editor) {
      // runs on page load, NOT on every keystroke in editor
      this.editor = editor
      if (this.cachedCode) {
        // TRACK? returned to lesson previously visited
        this.loadCodeFromCache()
        this.cachedStateMsg = "Pick up where you left off. We've saved your code for you!"
        if (this.lessonPassed) {
          this.run()
        }
      } else {
        // TRACK? first time starting lesson
      }
```
In the case of a file upload lesson specifically, if you revisit a previously passed lesson while you still have files uploaded in the browser, you get the result we'd expect. However, if your code is cached but your browser doesn't remember your uploaded files, when `run` runs to try to reproduce the state you left off in it errors out because it can't reference `files`: 

![image](https://user-images.githubusercontent.com/19171465/57036377-1ba50c00-6c22-11e9-9b3b-5ddd273b5ebf.png)

On first visit we deal with people not having uploaded files by disabling the "Submit" button and showing a reminder to upload files when you hover over it, but in this case the big red error is much more noticeable and you won't get to trying that.

Basically, there's no way to save the state that your code works to pass the lesson without actually having files accessible in the browser. 

Outside of file lessons, the place where this pattern of running `run` on load causes problems is when you ask the user to open the console to view something and they actually see it twice - once from page load and once from their submission. (Although we do this less frequently now that we have the logging feature.)

**Comment by terichadbourne** [comment]
@fsdiogo You might find that this is related to issue #132

**Comment by terichadbourne** [comment]
@fsdiogo Is this one you might have time to look into this week or next? Would be great to get it fixed before camp.

**Comment by fsdiogo** [comment]
I've glanced through this issue a while ago and it's not as simple to fix as it seems.

One way that is not ideal, is to disable the automatic submission of a lesson that is already passed. The downside is that you won't see the logs or output messages, and will have to click `Reset` and do it again to see them.

I thought about creating dummy files and submitting with those, but I'm not sure how's that gonna work with the file input and such.

I'll have to check it out.

**Comment by fsdiogo** [comment]
This has been fixed in https://github.com/ProtoSchool/protoschool.github.io/pull/267.

Try it out in [proto.school](https://proto.school/#/mutable-file-system/).
